modules/compiler/vecz/source/transform/packetizer.cpp

```diff
@@ -2778,37 +2809,48 @@ ValuePacket Packetizer::Impl::packetizeMemOp(MemOp &op) {
   return results;
 }
 
-ValuePacket Packetizer::Impl::packetizeMaskedAtomicRMW(
-    CallInst &CI, VectorizationContext::MaskedAtomicRMW AtomicInfo) {
+ValuePacket Packetizer::Impl::packetizeMaskedAtomic(
+    CallInst &CI, VectorizationContext::MaskedAtomic AtomicInfo) {
   ValuePacket results;
 
-  Value *const ptr = CI.getArgOperand(0);
-  Value *const val = CI.getArgOperand(1);
-  Value *const mask = CI.getArgOperand(2);
+  bool const IsCmpXchg = AtomicInfo.isCmpXchg();
 
-  assert(AtomicInfo.ValTy == val->getType() && "AtomicInfo mismatch");
-  auto const packetWidth = getPacketWidthForType(val->getType());
+  Value *const ptrArg = CI.getArgOperand(0);
+  Value *const valOrCmpArg = CI.getArgOperand(1);
+  Value *const maskArg = CI.getArgOperand(2 + IsCmpXchg);
```

Comment:
personally I'd prefer `isCmpXchg ? 3 : 2` as being more clear, but feel free to ignore

Reply:
Fair enough, yeah. I think I like this style well enough, though it's definitely subjective.

Reply:
I think to me I think of a bool as true or false, and I need to think a little to see it as a 1, but yes feel free to leave as is.